### PR TITLE
Changed enum derive store implementation.

### PR DESCRIPTION
**Pull request by orizi**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/3817)

### CODE REVIEW

File: crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types [hunk 1]
```diff
@@ -17,6 +17,28 @@ mod test_contract {
...
+        B: u16,
+    }
+
+    #[derive(Drop, Serde, starknet::Store)]
+    enum EnumWithDefault {
+        A: u8,
+        #[default]
+        B: u16,
+        C: u32,
+    }
```
Comment (LucasLvy) [inline crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types:32]:
```suggestion
    #[derive(Drop, Serde, starknet::Store, Default)]
    enum EnumWithDefault {
        A: u8,
        #[default]
        B: u16,
        C: u32,
    }
```
I think it should derive default also

File: crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types [hunk 2]
```diff
@@ -42,6 +64,28 @@ mod test_contract {
...
```
Comment (LucasLvy) [inline crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types:79]:
```suggestion
    #[derive(Drop, Serde, starknet::Store, Default)]
    enum EnumWithDefault {
        A: u8,
        #[default]
        B: u16,
        C: u32,
    }
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

**Review by orizi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/3817)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @ArielElp, @LucasLvy, and @spapinistarkware)

___
*[`crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types` line 32 at r1](https://reviewable.io/reviews/starkware-libs/cairo/3817#gh-1280858424:-Nam76n_7WnqE48mVind:bjivsyg) ([raw file](https://github.com/starkware-libs/cairo/blob/7726a55507ede9e9c21458874a719eec3f578ec3/crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types#L32)):*
<details><summary><i>Previously, LucasLvy (Lucas @ StarkWare) wrote…</i></summary><blockquote>

```suggestion
    #[derive(Drop, Serde, starknet::Store, Default)]
    enum EnumWithDefault {
        A: u8,
        #[default]
        B: u16,
        C: u32,
    }
```
I think it should derive default also
</blockquote></details>

that can happen only after `Default` derive exists.
___
*[`crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types` line 79 at r1](https://reviewable.io/reviews/starkware-libs/cairo/3817#gh-1280858797:-Nam7D-k2kgbaROnaQ_o:bjivsyg) ([raw file](https://github.com/starkware-libs/cairo/blob/7726a55507ede9e9c21458874a719eec3f578ec3/crates/cairo-lang-starknet/src/plugin/plugin_test_data/user_defined_types#L79)):*
<details><summary><i>Previously, LucasLvy (Lucas @ StarkWare) wrote…</i></summary><blockquote>

```suggestion
    #[derive(Drop, Serde, starknet::Store, Default)]
    enum EnumWithDefault {
        A: u8,
        #[default]
        B: u16,
        C: u32,
    }
```
</blockquote></details>

that can happen only after `Default` derive exists.

**Review by LucasLvy — APPROVED** [review]

**Review by spapinistarkware — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/3817#-:-NapjMxL8xNt5S8GkRdj:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @ArielElp and @LucasLvy)

**Review by orizi — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/3817)* status: all files reviewed, 2 unresolved discussions (waiting on @ArielElp and @LucasLvy)

**Review by LucasLvy — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/3817#-:-Napm06Q3202IZenuGgu:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @ArielElp)

**Pull request merged**